src/Mod/Fem/femsolver/run.py

```diff
@@ -50,6 +50,56 @@
 _dirTypes = {}
 
 
+def run_fem_solver(solver, working_dir=None):
+
+    if App.GuiUp:
+        import FreeCADGui
+        from PySide import QtGui
+
+    if solver.Proxy.Type == 'Fem::FemSolverCalculixCcxTools':
+        App.Console.PrintMessage("CalxuliX ccx tools solver!\n")
+        from femtools import ccxtools
+        fea = ccxtools.FemToolsCcx(None, solver)
+        fea.reset_mesh_purge_results_checked()
+        fea.run()
+    else:
+        App.Console.PrintMessage("Frame work solver!\n")
+        try:
+            if working_dir is not None:
+                machine = getMachine(solver, working_dir)
+            else:
+                machine = getMachine(solver)
+        except MustSaveError:
+            error_message = (
+                "Please save the file before executing the solver. "
+                "This must be done because the location of the working "
+                "directory is set to \"Beside .FCStd File\"."
```

Comment:
`\"Beside .FCStd File\"` 
What does this mean?

Reply:
in the directory of the FCStd file a new directory is created and inside this directory all files are created for the analysis.

Reply:
you can test in master ...
Start FreeCAD, open FEM Preferences, Tab General, There you have three possibilities to set the working dir for frame work solver. One of them is "Beside .FCStd File"

Reply:
any more comments on this luzpaz?